src/geocoding-mcp-server/Tools/GeocodingTool.cs

```diff
@@ -0,0 +1,101 @@
+using System.ComponentModel;
+using System.Text.Json;
+using ModelContextProtocol;
+using ModelContextProtocol.Server;
+
+namespace GeocodingMcpServer.Tools;
+
+/// <summary>
+/// MCP tool for geocoding addresses and landmarks to coordinates
+/// </summary>
+[McpServerToolType]
+public sealed class GeocodingTool
+{
+    private readonly ILogger<GeocodingTool> _logger;
+    
+    // Fallback coordinates for unknown locations (Rome city center)
+    private const double RomeCityCenterLatitude = 41.9028;
+    private const double RomeCityCenterLongitude = 12.4964;
+    
+    // Mock data for known landmarks and cities
+    // TODO: Consider externalizing to configuration file or database for easier maintenance
+    private static readonly Dictionary<string, (double Latitude, double Longitude)> KnownLocations = new(StringComparer.OrdinalIgnoreCase)
+    {
+        // Rome landmarks
+        { "colosseum", (41.8902, 12.4922) },
+        { "coliseum", (41.8902, 12.4922) },
+        { "roman forum", (41.8925, 12.4853) },
+        { "vatican", (41.9029, 12.4534) },
+        { "vatican city", (41.9029, 12.4534) },
+        { "pantheon", (41.8986, 12.4768) },
+        { "trevi fountain", (41.9009, 12.4833) },
+        { "spanish steps", (41.9058, 12.4823) },
+        { "trastevere", (41.8899, 12.4707) },
+        
+        // Cities
+        { "rome", (41.9028, 12.4964) },
+        { "roma", (41.9028, 12.4964) },
+        { "latina", (41.4677, 12.9037) },
+        
+        // Rome areas/neighborhoods
+        { "downtown rome", (41.9028, 12.4964) },
+        { "rome city center", (41.9028, 12.4964) },
+        { "termini station", (41.9008, 12.5015) }
+    };
+
+    public GeocodingTool(ILogger<GeocodingTool> logger)
+    {
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// Geocode an address or landmark to get its coordinates
+    /// </summary>
+    [McpServerTool]
+    [Description("Geocode an address or landmark to get its coordinates (latitude, longitude). Returns location details with coordinates.")]
+    public Task<string> GeocodeLocation(
+        [Description("Address or landmark name to geocode (e.g., 'Colosseum', 'Vatican', 'Rome')")] string location)
```

Comment:
given that knownlocations are in english, make it clear that the location parameter should be localized in english